smooth/components/component_electrolyzer_waste_heat.py

```diff
@@ -218,9 +218,9 @@ def electrolyzer_ratio_rule(model, t):
                 po.Constraint(model_to_solve.TIMESTEPS, rule=electrolyzer_ratio_rule)
                 )
 
-    def update_flows(self, results, sim_params):
+    def update_flows(self, results):
         # Check if the component has an attribute 'flows', if not, create it as an empty dict.
-        Electrolyzer.update_flows(self, results, sim_params, self.name)
+        Electrolyzer.update_flows(self, results, self.name)
         Electrolyzer.update_flows(
-            self, results, sim_params, self.name + "_thermal"
+            self, results, self.sim_params, self.name + "_thermal"
```

Comment:
Bug: Electrolyzer has no `update_flows`, so `Component.update_flows` is used. But this has no sim_params parameter any more.
Fix: `Electrolyzer.update_flows(self, results, self.name + "_thermal")`

Please check with a model that contains a waste heat electrolyzer.